src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1357 @@
+                `is_split_into_words=True` (to lift the ambiguity with a batch of sequences).
+            return_tensors (`str` or [`~utils.TensorType`], *optional*):
+                If set, will return tensors of a particular framework. Acceptable values are:
+                    - `'tf'`: Return TensorFlow `tf.constant` objects.
+                    - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                    - `'np'`: Return NumPy `np.ndarray` objects.
+                    - `'jax'`: Return JAX `jnp.ndarray` objects.
+
+        Returns:
+            [`BatchFeature`]: A [`BatchFeature`] with the following fields:
+
+            - **input_ids** -- List of token ids to be fed to a model. Returned when `text` is not `None`.
+            - **attention_mask** -- List of indices specifying which tokens should be attended to by the model (when
+              `return_attention_mask=True` or if *"attention_mask"* is in `self.model_input_names` and if `text` is not
+              `None`).
+            - **pixel_values** -- Pixel values to be fed to a model. Returned when `images` is not `None`.
+            - **pixel_mask** -- Pixel mask to be fed to a model. Returned when `images` is not `None`.
+        """
+        output_kwargs = self._merge_kwargs(
+            AriaProcessorKwargs,
+            {},
+            **kwargs,
+        )
+        if isinstance(text, str):
+            text = [text]
+        elif not isinstance(text, list) and not isinstance(text[0], str):
+            raise ValueError("Invalid input text. Please provide a string, or a list of strings")
+        if images is not None:
+            image_inputs = self.image_processor(
+                images,
+                return_tensors=output_kwargs["images_kwargs"]["return_tensors"],
+                max_image_size=output_kwargs["images_kwargs"]["max_image_size"],
+                split_image=output_kwargs["images_kwargs"]["split_image"],
+            )
+            # expand the image_token according to the num_crops and tokens per image
+            tokens_per_image = self.size_conversion[image_inputs.pixel_values.shape[2]]
+
+            prompt_strings = []
+            num_crops = image_inputs.pop("num_crops") * tokens_per_image
+            for sample in text:
+                sample = sample.replace(self.image_token, self.image_token * num_crops)
+                prompt_strings.append(sample)
+
+        else:
+            image_inputs = {}
+            prompt_strings = text
+
+        text_inputs = self.tokenizer(
+            prompt_strings,
+            return_tensors=output_kwargs["text_kwargs"]["return_tensors"],
+            padding=output_kwargs["text_kwargs"]["padding"],
+            truncation=output_kwargs["text_kwargs"]["truncation"],
+            max_length=output_kwargs["text_kwargs"]["max_length"],
+        )
+
+        return BatchFeature(data={**text_inputs, **image_inputs})
+
+    def save_pretrained(self, save_directory, **kwargs):
```

Comment:
no need to overwrite `from_pretrained` and `save_pretrained` unless the processor has extra special classes within like QformerProcessor in BLIP. I guess Aria is pretty much straightforward with inputs and should work without it